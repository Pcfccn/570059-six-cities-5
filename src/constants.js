const Path = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer/:id`,
  getUrlById: (id) => (`/offer/${id}`)
};

const ratingStars = [5, 4, 3, 2, 1];
const ratingInputTitle = [`perfect`, `good`, `not bad`, `badly`, `terribly`];

const widthStartsCoefficient = 20;

const CityCoordinates = {
  Amsterdam: [52.38333, 4.9],
  Paris: [48.864716, 2.349014],
};

const OfferCardClassName = {
  nearPlace: `near-places__card`,
  citiesPlace: `cities__place-card`,
};

const MapClassName = {
  property: `property__map`,
  cities: `cities__map`,
};

const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export {Path, widthStartsCoefficient, ratingStars, ratingInputTitle, CityCoordinates, OfferCardClassName, MapClassName, cities};
