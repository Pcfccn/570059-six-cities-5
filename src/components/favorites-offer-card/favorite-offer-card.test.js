import React from "react";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";
import browserHistory from "../../brouser-history";
import {AuthorizationStatus, SortType} from "../../constants";
import FavoritesOfferCard from "./favorite-offer-card";
import configureMockStore from 'redux-mock-store';

const getOffersMock = (count) => {
  const templateOffers = Array(count)
    .fill(``)
    .map((_, index) => {
      return {
        id: index + 1,
        city: `Amsterdam`,
        reviews: [index + 2134, index + 2135],
        image: [
          `/img/apartment-01.jpg`,
          `/img/apartment-02.jpg`,
          `/img/apartment-03.jpg`,
          `/img/room.jpg`,
          `/img/studio-01.jpg`,
          `/img/studio-photos.jpg`,
        ],
        isPremium: Boolean(Math.random()),
        isInBookmarks: Boolean(Math.random()),
        name: `Beautiful & luxurious studio`,
        type: `Apartament`,
        bedrooms: 3,
        adults: 4,
        rating: 3.5,
        price: {
          value: 1234,
          period: `night`,
        },
        inside: [
          `Wi-Fi`,
          `Washing machine`,
          `Towels`,
          `Heating`,
          `Coffee machine`,
          `Baby seat`,
          `Kitchen`,
          `Dishwasher`,
          `whatewer you want`,
        ],
        host: {
          avatar: `/img/avatar-angelina.jpg`,
          name: `Anagelina`,
          pro: Boolean(Math.random()),
          description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          The building is green and from 18th century.`,
        },
        location: [52.3909553943508, 4.85309666406198],
        previewImage: `/img/avatar-angelina.jpg`,
        cityLocation: [52.369553943508, 4.85309666406198],
        cityZoom: 10,
      };
    });

  if (count === 1) {
    const [oneOffer] = templateOffers;
    return oneOffer;
  }

  return templateOffers;
};

let mockState;
beforeEach(() => {
  mockState = {
    USER: {
      authorizationStatus: AuthorizationStatus.AUTH,
      email: `sdf@mail.ri`
    },
    STATE: {
      city: `Amsterdam`,
      enteredOffer: {
        location: getOffersMock(1).location,
        id: getOffersMock(1).id,
      },
      sortType: SortType.POPULAR_DESC,
    },
    DATA: {
      offers: getOffersMock(5)
    }
  };
});

it(`Should FavoritesOfferCard render correctly`, () => {
  const mockStore = configureMockStore();
  const store = mockStore(mockState);
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={browserHistory}>
            <FavoritesOfferCard
              favotiteOffer={getOffersMock(1)}
              offer={getOffersMock(1)}
            />          </Router>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
