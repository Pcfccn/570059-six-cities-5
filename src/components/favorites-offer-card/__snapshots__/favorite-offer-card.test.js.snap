// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesOfferCard render correctly 1`] = `
Array [
  <article
    className="favorites__card place-card"
  >
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="110"
          src="/img/avatar-angelina.jpg"
          width="150"
        />
      </a>
    </div>
    <div
      className="favorites__card-info place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            1234
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "70%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
        >
          Beautiful & luxurious studio
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Apartament
      </p>
    </div>
  </article>,
  "          ",
]
`;
